plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java

```diff
@@ -47,11 +63,57 @@ public DockerConnectorConfiguration(URI dockerDaemonUri,
     }
 
     private static URI dockerDaemonUri() {
-        return SystemInfo.isLinux() ? DockerConnector.UNIX_SOCKET_URI : DockerConnector.BOOT2DOCKER_URI;
+        if (SystemInfo.isLinux()) {
+            return UNIX_SOCKET_URI;
+        }
+
+        // check if have boot2docker variables
+        String host = System.getenv(BOOT2DOCKER_HOST_PROPERTY);
+        if (host != null) {
+            // user has defined some properties, use them
+            URI userURI;
+            try {
+                userURI = new URI(host);
+            } catch (URISyntaxException e) {
+                LOG.error(String.format("Unable to parse {0} property with value {1}", BOOT2DOCKER_HOST_PROPERTY, host), e);
+                // unable to use given property, fallback to default URL
+                return DEFAULT_BOOT2DOCKER_URI;
+            }
+
+            // Secure connection ?
+            String tls = System.getenv(BOOT2DOCKER_TLS_VERIFY_PROPERTY);
+            String protocol = "1".equals(tls) ? "https" : "http";
+
+            // build URI
+            try {
+                return new URI(protocol, null,  userURI.getHost(), userURI.getPort(), null, null, null);
+            } catch (URISyntaxException e) {
+                LOG.error(String.format("Unable to create URI {0} property with value {1} and TLS {2}", BOOT2DOCKER_HOST_PROPERTY, host, BOOT2DOCKER_TLS_VERIFY_PROPERTY), e);
+                // unable to use given property, fallback to default URL
+                return DEFAULT_BOOT2DOCKER_URI;
+
+            }
+        }
+
+
+
+
+
```

Comment:
Generally it is okay, but come on is it really necessary to have 5 empty lines here?